_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,227 @@
+---
+name: iframe has a unique accessible name
+
+description: |
+ The accessible name for each iframe is unique
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to `iframe` elements that are [exposed to assistive technologies](#exposed-to-assistive-technologies) and have a [non-empty](#non-empty) [accessible name](#accessible-name).
+
+### Expectation
+
+Each target element has an [accessible name](#accessible-name) that is unique within all applicable `iframes` within the same [document tree](https://www.w3.org/TR/dom41/#document-trees) or [shadow tree](https://www.w3.org/TR/dom41/#shadow-trees), where the `src` attributes are different.
```

Comment:
I'm not entirely sure about the phrasing "unique within". From my understanding of English, `A` is unique _within_ `B` if `B` includes only one occurrence of `A`, e.g. the word "ipsum" is unique within the sentence "lorem ipsum dolor". On the other hand, `A` if unique _across_ `X`, `Y`, and `Z` if only one of them is equal to `A`, e.g. "ipsum" is unique across the words "lorem", "ipsum", "dolor". It seems to me that we're looking for the latter, i.e. "[...] has an accessible name that is unique across accessible names for all applicable iframes [...]".

It would be great if someone native to the language could set this straight though 😄